plugins/e57/io/E57Reader.cpp

```diff
@@ -33,14 +33,48 @@
 ****************************************************************************/
 
 #include <iostream>
+#include <E57Format.h>
 #include <E57Exception.h>
 
 #include "E57Reader.hpp"
+#include "Scan.hpp"
 #include "Utils.hpp"
 
 namespace pdal
 {
 
+class E57Reader::ChunkReader
```

Comment:
Do we really need this class?
I believe its always better to have specific class to handle specific class.
But I think we can handle things better without this class, Please check #2757 .

Reply:
Sorry, I didn't see your PR linked in #2707 where I started from.
My goal was to change as little as possible in order to remove the E57 stuff from header files only. So these classes have only been moved to the implementation and are unchanged otherwise. But if I were to completely reimplement these plugins I would probably remove them, yes.